crates/oxc_linter/src/disable_directives.rs

```diff
@@ -164,10 +164,20 @@ impl<'a> DisableDirectivesBuilder<'a> {
                 // `eslint-disable-next-line`
                 else if let Some(text) = text.strip_prefix("-next-line") {
                     // Get the span up to the next new line
-                    let stop = source_text[comment_span.end as usize..]
-                        .lines()
-                        .take(2)
-                        .fold(comment_span.end, |acc, line| acc + line.len() as u32);
+                    let mut stop = comment_span.end;
+                    let mut lines_after_comment_end =
+                        source_text[comment_span.end as usize..].split_inclusive('\n');
+
+                    if let Some(rest_of_line) = lines_after_comment_end.next() {
+                        stop += rest_of_line.len() as u32;
+                    }
+
+                    if let Some(next_line) = lines_after_comment_end.next() {
+                        let next_line = next_line.strip_suffix('\n').unwrap_or(next_line);
```

Comment:
These APIs accept a pattern https://doc.rust-lang.org/std/str/pattern/index.html

So a ['\n', '\r'] should be possible.

And it seems like you can use trim_end_matches? https://doc.rust-lang.org/std/primitive.str.html#method.trim_end_matches